packages/relay/src/lib/clients/sdkClient.ts

```diff
@@ -781,22 +721,28 @@ export class SDKClient {
       return transactionResponse;
     } catch (e: any) {
       if (e instanceof JsonRpcError) {
+        this.logger.warn(
+          e,
+          `${requestDetails.formattedRequestId} Failed to execute ${txConstructorName} transaction due to a JsonRpcError: transactionId=${transaction.transactionId}, callerName=${callerName}, status=${e.code} message=${e.message}`,
+        );
         throw e;
       }
```

Comment:
QQ: What's the point of logging a warning here and throwing the error?

Reply:
I think we should have a stack trace inside of the error that is being thrown and this warning log would only clutter the logs with duplicated errors, correct me if I'm wrong

Reply:
Yes, you're correct that the error includes a stack trace. However, that trace is only visible if explicitly logged. If a higher-level handler catches the error and performs logic without logging it, no log would be generated, leading to a loss of context and detail. This lack of logging can make debugging more challenging, as the original context of the error isn't captured. Logging a warning here ensures the relevant details are preserved, even if the error handling at higher levels omits logging.

Reply:
> If a higher-level handler catches the error and performs logic without logging it, no log would be generated, leading to a loss of context and detail.

Usually that's a responsibility of the piece of code which handles the error - they can choose to either re-throw the error or log a warning and ignore it or do something else. Logging a warning before every error, can easily lead to many duplicated logs and make it harder to trace the application flow.

If you insist on keeping this, let's change it from a `warn` to a `trace` log, so that these "safety/traceability logs" are shown only on `trace` level.

Reply:
@victor-yanev I would agree with Victor. It is responsibility of the handler to decide what to do. I think we shouldn't clutter the code with logs, since it can affect memory and performance and lead to confusion when investigating an error through logs

Reply:
Updated in https://github.com/hashgraph/hedera-json-rpc-relay/pull/3229/commits/9f9f1622d8b4e3c692abe19ae86e5ca80399e08e. Thanks for the suggestion!